app/javascript/githubRepos/singleRepo.jsx

```diff
@@ -0,0 +1,75 @@
+import { h, Component } from 'preact';
+import PropTypes from 'prop-types';
+
+export class SingleRepo extends Component {
+  constructor(props) {
+    super(props);
+    const { selected } = this.props;
+    this.state = { selected };
+  }
+
+  forkLabel = () => {
+    const { fork } = this.props;
+    if (fork) {
+      return <span className="github-repo-fork">fork</span>;
+    }
+    return null;
+  };
+
+  submitRepo = () => {
+    const { selected } = this.state;
+    const { githubIdCode } = this.props;
+    this.setState({ selected: !selected });
```

Comment:
I added optimistic rendering here by changing the state initially on the first click, and then the async fetch will come back and change it as necessary. As seen in the gif, spamming the button would show some weird behavior. Not sure how to fix that. Also, I should probably put in a rate limit somewhere.